source/common/stats/stats_impl.h

```diff
@@ -182,54 +203,32 @@ struct RawStatData {
   ~RawStatData() = delete;
 
   /**
-   * Configure static settings. This MUST be called
-   * before any other static or instance methods.
-   */
-  static void configure(Server::Options& options);
-
-  /**
-   * Allow tests to re-configure this value after it has been set.
-   * This is unsafe in a non-test context.
-   */
-  static void configureForTestsOnly(Server::Options& options);
-
-  /**
-   * Returns the maximum length of the name of a stat. This length
-   * does not include a trailing NULL-terminator.
-   */
-  static size_t maxNameLength() { return maxObjNameLength() + MAX_STAT_SUFFIX_LENGTH; }
-
-  /**
-   * Returns the maximum length of a user supplied object (route/cluster/listener)
-   * name field in a stat. This length does not include a trailing NULL-terminator.
-   */
-  static size_t maxObjNameLength() {
-    return initializeAndGetMutableMaxObjNameLength(DEFAULT_MAX_OBJ_NAME_LENGTH);
-  }
-
-  /**
-   * Returns the maximum length of a stat suffix that Envoy generates (over the user supplied name).
-   * This length does not include a trailing NULL-terminator.
+   * Returns the size of this struct, accounting for the length of name_
+   * and padding for alignment. Required for the HeapRawStatDataAllocator, which does not truncate
+   * at a maximum stat name length.
    */
-  static size_t maxStatSuffixLength() { return MAX_STAT_SUFFIX_LENGTH; }
+  static uint64_t sizeGivenName(const std::string& name);
```

Comment:
name could be specified as absl::string_view with no perf or compatibility downside here, I think.

Reply:
I like this idea. A lot of the complexity of this ~80-file change is having to inject stats_options into any place that creates a Cluster (Route, Listener, etc.) because of name length limitations. (See https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/cds.proto#L46-L55). I guess the fear here is that if we don't truncate at cluster naming time, we might truncate later, and a cluster named `foo[...many characters...]1` and a cluster named `foo[..many characters...]2` might end up emitting identical stats?